internal/host/static/query.go

```diff
@@ -45,5 +45,172 @@ final (action, host_id) as (
 )
 select * from final
 order by action, host_id;
+`
+
+	estimateCountHosts = `
+select sum(reltuples::bigint) as estimate from pg_class where oid in ('static_host'::regclass)
+`
+
+	listHostsTemplate = `
+with hosts as (
+    select public_id,
+           create_time,
+           update_time,
+           name,
+           description,
+           catalog_id,
+           address,
+           version
+      from static_host
+     where catalog_id = @catalog_id
+  order by create_time desc, public_id asc
+     limit %d
+),
+host_set_ids as (
+    select string_agg(distinct set_id, '|') as set_ids,
+           host_id
+      from static_host_set_member
+     where host_id in (select public_id from hosts)
+  group by host_id
+),
+final as (
+           select h.public_id,
+                  h.create_time,
+                  h.update_time,
+                  h.name,
+                  h.description,
+                  h.catalog_id,
+                  h.address,
+                  h.version,
+                  hsi.set_ids
+             from hosts h
+  left outer join host_set_ids hsi on hsi.host_id = h.public_id
+)
+  select *
+    from final
+order by create_time desc, public_id asc;
```

Comment:
This query is based on the [`static_host_with_set_memberships`](https://github.com/hashicorp/boundary/blob/600a2dff240a7a46c3b420b0ed3bd29c50e76f41/internal/db/schema/migrations/oss/postgres/25/01_static_host_view.up.sql#L9-L23) view.